### PR TITLE
chore(docs): restructure backup/recovery documentation

### LINKED ISSUE #2536: Reorganize the backup and recovery section in the docs
Looking at the current documentation for backup and recovery in light of adding support for Kubernetes volume snapshots (#2081), I am proposing the following changes to the current structure:

- Split “Backup” from “Recovery”
- In the “Backup” page:
    - Explain the two methods: object store and volume snapshots
    - Explain the role of WAL archiving for both cold/hot backups and PITR
    - Remove all the cloud provider stuff and move it in an appendix
- In the “Recovery” page we:
    - Explain the two sources of recovery: object store and volume snapshots
    - Provide an overview of both solutions (see example table below)
    - Describe both solutions, with examples, and cover PITR as well as snapshot recovery
- Refactor content in the bootstrap section or make sure they are aligned
- Any other change that emerges and makes sense

The goal of this ticket is to simply move the current documentation in the different section, and maybe add placeholders for the future rewriting in issues #2387 and #2527, without adding any content. This will make it clear to distinguish the specific content of the documentation for the previously mentioned tickets.

I have drafted an initial matrix of the possibilities that volume snapshots will bring to the game, for future memory:

|                                      | WAL archiving | Cold backup | Hot backup | Backup from a standby | Snapshot recovery | Point In Time Recovery (PITR) |
|--------------------------------------|:-------------:|:-----------:|:----------:|:---------------------:|:-----------------:|:-----------------------------:|
| Object store                         |      Yes      |      No     |     Yes    |          Yes          |        No*        |              Yes              |
| Volume Snapshots without WAL archive |       No      |     Yes     |     No     |          Yes          |        Yes        |               No              |
| Volume Snapshots with WAL archive    |      Yes      |     Yes     |     Yes    |          Yes          |        Yes        |              Yes              |

### PR DESCRIPTION
**Pull request by gbartolini**
Restructure the existing documentation on backup and recovery to prepare for the introduction of volume snapshot support.

Closes #2536

### CODE REVIEW

**Review by armru — APPROVED** [review]

File: docs/src/appendixes/object_stores.md [hunk 1]
```diff
@@ -0,0 +1,424 @@
+# Appendix A - Common object stores
```
Comment (djw-m) [inline docs/src/appendixes/object_stores.md:1]:
```suggestion
# Appendix A - Common object stores for backups
```

File: docs/src/appendixes/object_stores.md [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+
+You can archive the backup files in any service that is supported
```
Comment (djw-m) [inline docs/src/appendixes/object_stores.md:3]:
```suggestion
You can store the [backup](../backup) files in any service that is supported
```

File: docs/src/appendixes/object_stores.md [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+- [AWS S3](https://aws.amazon.com/s3/)
+- [Microsoft Azure Blob Storage](https://azure.microsoft.com/en-us/services/storage/blobs/)
+- [Google Cloud Storage](https://cloud.google.com/storage/)
+
+You can also use any compatible implementation of the supported services.
+
+The required setup depends on the chosen storage provider and is
+discussed in the following sections.
+
+## S3
```
Comment (djw-m) [inline docs/src/appendixes/object_stores.md:15]:
```suggestion
## AWS S3
```

File: docs/src/appendixes/object_stores.md [hunk 4]
```diff
@@ -0,0 +1,424 @@
+# Appendix A - Common object stores
+
+You can archive the backup files in any service that is supported
+by the Barman Cloud infrastructure. That is:
+
+- [AWS S3](https://aws.amazon.com/s3/)
```
Comment (djw-m) [inline docs/src/appendixes/object_stores.md:6]: Suggest making these links to the section headings below and in the sections link out to the product pages. eg

* [AWS S3](#aws_s3)

## AWS S3

Amazon's [S3](https://aws.amazon.com/s3/) object storage allows... (some general notes/guidance)....

Comment (gbartolini) [reply docs/src/appendixes/object_stores.md]: Done

**Review by djw-m — APPROVED** [review]
Tested under mkdocs, LGTM

**Pull request merged**